cfn-resources/network-peering/cmd/resource/resource.go

```diff
@@ -102,28 +98,26 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		_, _ = logger.Debugf("AwsAccountIdwas not set, default to req.RequestContext.AccountID:%v", awsAccountID)
 	}
 
-	peerRequest := mongodbatlas.Peer{
-		ContainerID:         *currentModel.ContainerId,
-		VpcID:               *currentModel.VpcId,
-		AccepterRegionName:  *currentModel.AccepterRegionName,
-		AWSAccountID:        *awsAccountID,
-		RouteTableCIDRBlock: *currentModel.RouteTableCIDRBlock,
-		ProviderName:        constants.AWS,
+	peerRequest := admin.BaseNetworkPeeringConnectionSettings{
+		ContainerId:         *currentModel.ContainerId,
+		VpcId:               currentModel.VpcId,
+		AccepterRegionName:  currentModel.AccepterRegionName,
+		AwsAccountId:        awsAccountID,
+		RouteTableCidrBlock: currentModel.RouteTableCIDRBlock,
+		ProviderName:        admin.PtrString(constants.AWS),
 	}
 
-	peerResponse, resp, err := client.Peers.Create(context.Background(), projectID, &peerRequest)
+	peerResponse, resp, err := client.AtlasV2.NetworkPeeringApi.CreatePeeringConnection(context.Background(), projectID, &peerRequest).Execute()
 	if err != nil {
-		_, _ = logger.Warnf("error creating network peering: %s", err)
-		return progressevents.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
-			resp.Response), nil
+		return progressevent.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
+			resp), nil
```

Comment:
I normally prefer to use the default message if we're not adding info:  		return progressevent.GetFailedEventByResponse(err.Error(), resp), nil

Reply:
updated